### PR TITLE
itest+lntemp: add `ChainKitClient` RPCs and refactor `testChainKit`

### PR DESCRIPTION
**Pull request by yyforyongyu**
This PR adds the newly added `ChainKitClient` to `HarnessRPC`, which can be used in other tests.

### CODE REVIEW

File: lntest/itest/lnd_onchain_test.go [hunk 1]
```diff
@@ -24,34 +24,28 @@ import (
...
 
 	// Retrieve the best block by hash.
-	getBlockRes, err := ht.Alice.RPC.ChainKit.GetBlock(
-		ctx, &chainrpc.GetBlockRequest{
-			BlockHash: bestBlockHash.CloneBytes(),
-		},
-	)
-	require.NoError(ht, err)
+	getBlockReq := &chainrpc.GetBlockRequest{
+		BlockHash: bestBlockHash.CloneBytes(),
```
Comment (guggero) [inline lntest/itest/lnd_onchain_test.go:46]: nit: could just be `bestBlockHash[:]`.

File: lntemp/rpc/chain_kit.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	defer cancel()
+
+	resp, err := h.ChainKit.GetBlock(ctxt, req)
+	h.NoError(err, "GetBlock")
+
+	return resp
+}
+
+// GetBlockHash makes a RPC call to chain kit client's GetBlockHash and asserts.
+func (h *HarnessRPC) GetBlockHash(
```
Comment (guggero) [inline lntemp/rpc/chain_kit.go:48]: Nice! I forgot to add a comment about adding those helper function in the original PR...

**Review by guggero — APPROVED** [review]
Very nice, thanks for the cleanup! LGTM :tada:

File: lntest/itest/lnd_onchain_test.go (hunk 1 above)
Comment (yyforyongyu) [reply lntest/itest/lnd_onchain_test.go]: Switched!

File: lntemp/rpc/chain_kit.go (hunk 2 above)
Comment (yyforyongyu) [reply lntemp/rpc/chain_kit.go:48]: Maybe I should put some notes about how to add new RPC clients/endpoints in the `lntest`'s readme...

**Review by ffranr — APPROVED** [review]
Looks good!

ntACK -- I haven't run these tests. Could you help me with an example command for running the new temp tests please?

File: lntemp/rpc/chain_kit.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	"context"
+
+	"github.com/lightningnetwork/lnd/lnrpc/chainrpc"
+)
+
+// =====================
+// ChainKitClient related RPCs.
+// =====================
+
+// GetBestBlock makes a RPC call to chain kit client's GetBestBlock and asserts.
```
[the remark this answers is not in the document]
Comment (yyforyongyu) [reply lntemp/rpc/chain_kit.go]: fixed!

**Comment by yyforyongyu** [comment]
@ffranr to run the temp case, atm, use `make itest temptest=true icase=$icase`. Eventually, we will be using `make itest icase=$icase` tho.

**Comment by ffranr** [comment]
> @ffranr to run the temp case, atm, use `make itest temptest=true icase=$icase`. Eventually, we will be using `make itest icase=$icase` tho.

Thanks! Ran the tests and all seemed fine.

**Pull request merged**